plugins/woocommerce/src/Caches/OrderCacheController.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Automattic\WooCommerce\Caches;
+
+/**
+ * A class to control the usage of the orders cache.
+ */
+class OrderCacheController {
+
+	public const ORDERS_CACHE_USAGE_ENABLED_OPTION = 'woocommerce_orders_cache_enabled';
+
+	public const ORDERS_CACHE_USAGE_ENABLED_BACKUP_OPTION = 'woocommerce_orders_cache_enabled_backup';
+
+	/**
+	 * The orders cache to use.
+	 *
+	 * @var OrderCache
+	 */
+	private $order_cache;
+
+	/**
+	 * Class initialization, invoked by the DI container.
+	 *
+	 * @internal
+	 * @param OrderCache $order_cache The order cache engine to use.
+	 */
+	final public function init( OrderCache $order_cache ) {
+		$this->order_cache = $order_cache;
+	}
+
+	/**
+	 * Set the value of the order cache usage setting.
+	 *
+	 * @param bool $enable True if the order cache should be used, false if not.
+	 * @throws \Exception Attempt to enable the orders cache usage while it's temporarily disabled.
+	 */
+	public function set_orders_cache_usage( bool $enable ): void {
+		if ( $enable && $this->orders_cache_usage_is_temporarly_disabled() ) {
+			throw new \Exception( "OrderCacheController::set_orders_cache_usage: cache usage can't be enabled while it's temporarily disabled, run maybe_restore_orders_cache_usage first." );
+		}
+
+		$this->order_cache->flush();
+		update_option( self::ORDERS_CACHE_USAGE_ENABLED_OPTION, $enable ? 'yes' : 'no' );
+	}
+
+	/**
+	 * Get the value of the order cache usage setting.
+	 *
+	 * @return bool True if order cache usage setting is currently enabled, false if not.
+	 */
+	public function orders_cache_usage_is_enabled(): bool {
+		return 'yes' === get_option( self::ORDERS_CACHE_USAGE_ENABLED_OPTION );
+	}
+
+	/**
+	 * Temporarily disable the order cache if it's enabled.
+	 *
+	 * A backup option is created with the current value of the cache usage enable setting,
+	 * and then the setting is disabled. If the backup option already exists nothing is done.
+	 * The original setting value can be restored with maybe_restore_orders_cache_usage.
+	 */
+	public function temporarily_disable_orders_cache_usage(): void {
+		$this->order_cache->flush();
+		if ( $this->orders_cache_usage_is_temporarly_disabled() ) {
+			return;
+		}
+
+		update_option( self::ORDERS_CACHE_USAGE_ENABLED_BACKUP_OPTION, get_option( self::ORDERS_CACHE_USAGE_ENABLED_OPTION ) );
```

Comment:
if we are disabling it temporarily, it should probably not be persistently stored in an option. the use case of this method is to disable it in a request, or in a corn job for example. so maybe we can store this in a memory and it should be cleared whenever the request/cron ends.

Reply:
Makes sense. I'll switch from an option to a property inside the controller class (which is single-instance).